components/automate-ui/src/app/services/projects-filter/projects-filter.effects.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { of as observableOf } from 'rxjs';
+import { switchMap, catchError, map, tap } from 'rxjs/operators';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { ProjectsFilterOption } from './projects-filter.reducer';
+import { ProjectsFilterService } from './projects-filter.service';
+import {
+  ProjectsFilterActionTypes,
+  LoadOptions,
+  LoadOptionsSuccess,
+  LoadOptionsFailure,
+  SaveOptions
+} from './projects-filter.actions';
+
+@Injectable()
+export class ProjectsFilterEffects {
+  constructor(
+    private actions$: Actions,
+    private projectsFilter: ProjectsFilterService
+  ) { }
+
+  @Effect()
+  loadOptions$ = this.actions$.pipe(
+    ofType<LoadOptions>(ProjectsFilterActionTypes.LOAD_OPTIONS),
+    switchMap(() => {
+      return this.projectsFilter.fetchOptions().pipe(
+        map((fetched: ProjectsFilterOption[]) => {
```

Comment:
What does this code do (L29-33) ?

Reply:
At the moment this PR has the projectOptions and their `checked` state being stored in localstorage to persist across page refreshes, sessions, etc.

This code is grabbing the previously saved projectOptions from localstorage (`restored`), and if any of them has the same `value` as one of the newly fetched projectOptions (`fetched`), its `checked` status is merged with it to create the final list of available options (`loaded`).

Reply:
Nice! Thanks for the details!